google/cloud/aiplatform/base.py

```diff
@@ -334,6 +343,9 @@ def display_name(self) -> str:
         """Display name of this resource."""
         return self._gca_resource.display_name
 
+    def __repr__(self) -> str:
+        return repr(self._gca_resource)
```

Comment:
Will we eventually replace this with a prettier representation when time permits?

Another concern is users trying to access fields shown in `_gca_resource`, for example:

```python
ds = aiplatform.Dataset('7529226928591470592')
ds
```

```
name: "projects/580378083368/locations/us-central1/datasets/7529226928591470592"
display_name: "test_dataset_display_name"
metadata_schema_uri: "gs://google-cloud-aiplatform/schema/dataset/metadata/image_1.0.0.yaml"
create_time {
  seconds: 1617354353
  nanos: 956444000
}
update_time {
  seconds: 1617354361
  nanos: 707254000
}
etag: "AMEw9yMYcQqy2DwtHCx2ObYHK52UFD7YsBNGZnYRA-d7Fnau376JsgwTnBkrqtBb8CXk"
labels {
  key: "aiplatform.googleapis.com/dataset_metadata_schema"
  value: "IMAGE"
}
metadata {
  struct_value {
    fields {
      key: "dataItemSchemaUri"
      value {
        string_value: "gs://google-cloud-aiplatform/schema/dataset/dataitem/image_1.0.0.yaml"
      }
    }
    fields {
      key: "gcsBucket"
      value {
        string_value: "cloud-ai-platform-62f88573-082b-416d-a1b7-e47dcc520767"
      }
    }
  }
}
```

```python
ds.labels

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-4-7ef790277785> in <module>
----> 1 ds.labels

AttributeError: 'Dataset' object has no attribute 'labels'
```